lib/index.ts

```diff
@@ -21,26 +21,58 @@ async function setup(devicePath: string): Promise<express.Application> {
 	 */
 	app.post(
 		'/dut/on',
-		async (req: express.Request, res: express.Response): Promise<void> => {
-			res.send('OK');
-			await worker.on();
+		async (
+			_req: express.Request,
+			res: express.Response,
+			next: express.NextFunction,
+		) => {
+			try {
+				await worker.on();
+				res.send('OK');
+			} catch (err) {
+				next(err);
+			}
 		},
 	);
-	app.post('/dut/off', async (req: express.Request, res: express.Response) => {
-		res.send('OK');
-		await worker.off();
+	app.post(
+		'/dut/off',
+		async (
+			_req: express.Request,
+			res: express.Response,
+			next: express.NextFunction,
+		) => {
+			try {
+				await worker.off();
+				res.send('OK');
+			} catch (err) {
+				next(err);
+			}
+		},
+	);
+	app.use(async function(
+		err: Error,
+		_req: express.Request,
+		res: express.Response,
+		_next: express.NextFunction,
+	) {
+		res.status(500).send(err.message);
 	});
+
 	app.post(
 		'/dut/flash',
 		async (req: express.Request, res: express.Response) => {
 			const timer = setInterval(() => {
-				res.write('Still Flashing');
+				res.status(202).write('Still Flashing');
```

Comment:
I think you should set the status only once.

Reply:
Moved the code in the try block.